fabric-input-events-v1/src/main/java/net/fabricmc/fabric/impl/client/input/FabricMouseImpl.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.input;
+
+import org.lwjgl.glfw.GLFW;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.util.Window;
+
+import net.fabricmc.fabric.api.client.input.v1.FabricKeyboard;
+
+public class FabricMouseImpl {
+	public static FabricMouseImpl INSTANCE = new FabricMouseImpl();
+
+	private double x = 0.0;
+	private double y = 0.0;
+	private int buttons = 0;
+	private int modKeys = 0;
+	private final double[] current_x = new double[1];
+	private final double[] current_y = new double[1];
+
+	private FabricMouseImpl() {
+	}
+
+	public double getX() {
+		return this.x;
+	}
+
+	public double getY() {
+		return this.y;
+	}
+
+	public int getPressedButtons() {
+		return this.buttons;
+	}
+
+	public boolean isButtonPressed(int button) {
+		return (this.buttons & (1 << button)) != 0;
+	}
+
+	public int getModKeys() {
+		return this.modKeys;
+	}
+
+	public void update() {
+		MinecraftClient client = MinecraftClient.getInstance();
+
+		if (client == null) {
+			return;
+		}
+
+		Window window = client.getWindow();
+		long handle = window.getHandle();
+		GLFW.glfwGetCursorPos(handle, current_x, current_y);
```

Comment:
I suggest using FloatBuffers with the MemoryStack API instead of arrays https://github.com/LWJGL/lwjgl3-wiki/wiki/1.3.-Memory-FAQ#1-stack-allocation